src/number_theory/primes_congruent_one.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2020 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Riccardo Brasca
+-/
+
+import ring_theory.polynomial.cyclotomic
+import topology.algebra.polynomial
+import field_theory.finite.basic
+
+/-!
+# Primes congruent to one
+
+We prove that, for any positive `k : ℕ`, there are infinitely many primes `p` such that
+`p ≡ 1 [MOD k]`.
+-/
+
+namespace polynomial
+
+open nat
+
+/-- For any positive `k : ℕ` there are infinitely many primes `p` such that `p ≡ 1 [MOD k]`. -/
+lemma exists_infinite_primes_cong_one (k n : ℕ) (hpos : 0 < k) :
+  ∃ (p : ℕ), nat.prime p ∧ n ≤ p ∧ p ≡ 1 [MOD k] :=
```

Comment:
I think that this lemma should not be be in the namespace `polynomial`. I'd place it in the namespace `nat`.
This lemma should be named `nat.exists_prime_ge_modeq_one`. It would be nice to have lemmas `frequently_at_top_modeq_one : ∃ᶠ p in at_top, nat.prime p ∧ p ≡ 1 [MOD k]` and `infinite_set_of_prime_cong_one : set.infinite {p : ℕ | nat.prime p ∧ p ≡ 1 [MOD k]}`. For the first one you can use `filter.frequently_at_top`, and for the second it makes sense to add this at the bottom of `order/filter/cofinite`:

```
lemma nat.frequently_at_top_iff_infinite {p : ℕ → Prop} :
  (∃ᶠ n in at_top, p n) ↔ set.infinite {n | p n} :=
by simp only [← nat.cofinite_eq_at_top, frequently_cofinite_iff_infinite]
```

Reply:
Ah yes, I switched `nat` and `polynomial`!